data_node/upload_manager.go

```diff
@@ -128,63 +133,62 @@ func (um *UploadManager) handleAppendUpload(w http.ResponseWriter, r *http.Reque
 	id := r.Header.Get("ID")
 	if !um.validateIDExistance(id) {
 		log.Println(um.logPrefix, r.RemoteAddr, "ID not found")
-		handleRequestError(w, http.StatusNotFound, "ID not found")
+		handleRequestError(w, http.StatusForbidden, "ID not found")
 		return
 	}
 
-	chunkSize, err := strconv.ParseInt(r.Header.Get("Content-Length"), 10, 64)
-	if errors.IsError(err) || chunkSize <= 0 {
+	contentLength, err := strconv.ParseInt(r.Header.Get("Content-Length"), 10, 64)
```

Comment:
I have a question, at line 119, you say `r.ContentLength`, is that different from the value in the `Content-Length`. I mean does it actually calculate the request size or does it check this header?